node/nodeRunner.go

```diff
@@ -1434,6 +1436,9 @@ func (nr *nodeRunner) CreateManagedNetworkComponents(
 	if nr.configs.PreferencesConfig.Preferences.FullArchive {
 		networkComponentsFactoryArgs.NodeOperationMode = common.FullArchiveMode
 	}
+	if nr.configs.PreferencesConfig.Preferences.LightClient {
```

Comment:
with this order, we force that if both full archive and light client is set, the node will be light client. I think it is ok, but perhaps we should add some restrictions on config values, to avoid the bad config situation when both operation modes are on.

@AdoAdoAdo what do you think?

Reply:
perhaps add a todo for further PR, it may need bigger changes as networkMessenger only accepts one operation mode at this time

maybe we can use bit operations to replace the NodeOperationMode and use FullArchive &  LightClient

Reply:
as discussed during meet, perhaps a slice of strings would be a better approach